sklearn/utils/estimator_checks.py

```diff
@@ -1113,6 +1113,38 @@ def check_array_api_input(
         "transform",
     )
 
+    try:
+        np.asarray(X_xp)
+        np.asarray(y_xp)
+        # TODO There are a few errors in SearchCV with array-api-strict because
+        # we end up doing X[train_indices] where X is a array-api-strict array
+        # and indices a numpy array. Probably not worth investigating for now,
+        # since using array-api-strict with array API disabled does not seem a
+        # very relevant.
```

Comment:
While I agree that we don't really care about supporting `array-api-strict` inputs when array API dispatch is disabled, I still think that those `X[train_indices]` are bugs in our code. We should probably always do `X[xp.asarray(train_indices, device=device(X)]` in our CV tools.

Reply:
I tried to tweak the comment a bit to be more self-explanatory. I am not sure it is so easy to tackle and discussing this in real-life this is probably not worth doing anything about it for now. All the array API library I could try (PyTorch, jax, CuPy) accept to index with numpy array.

For compleness, the proposed work-around would not work because the issue is that `X` is an array-api-strict array, `train_indices` is a numpy array and array-api-strict is insisting that to index `X` you need to pass array-api-strict array.